config/config.go

```diff
@@ -83,134 +100,247 @@ type CircuitBreaker struct {
+				ReadHeader: 2 * time.Second,
+				Write:      5 * time.Second,
+				Idle:       20 * time.Second,
+				Handler:    5 * time.Second,
+			},
+			Forwarding: Forward{
+				HTTP2HTTPS:         true,
+				RedirectStatusCode: 301,
+			},
+			GZip: false,
+		},
+		Cache: Cache{
+			Port:            "6379",
+			DB:              0,
+			TTL:             0,
+			AllowedStatuses: []int{200, 301, 302},
+			AllowedMethods:  []string{"HEAD", "GET"},
+		},
+		CircuitBreaker: CircuitBreaker{
+			Threshold:   2,                // after 2nd request, if meet FailureRate goes open.
+			FailureRate: 0.5,              // 1 out of 2 fails, or more
+			Interval:    0,                // doesn't clears counts
+			Timeout:     60 * time.Second, // clears state after 60s
+			MaxRequests: 1,
+		},
 	}
+}
 
-	return value
+func getEnvConfig() Configuration {
+	return Configuration{
+		Server: Server{
+			Port: Port{
+				HTTP:  utils.GetEnv("SERVER_HTTP_PORT", ""),
+				HTTPS: utils.GetEnv("SERVER_HTTPS_PORT", ""),
+			},
+			TLS: TLS{
+				Auto:     utils.GetEnv("TLS_AUTO_CERT", "") == "1",
+				Email:    utils.GetEnv("TLS_EMAIL", ""),
+				CertFile: utils.GetEnv("TLS_CERT_FILE", ""),
+				KeyFile:  utils.GetEnv("TLS_KEY_FILE", ""),
+			},
+			Timeout: Timeout{
+				Read:       utils.ConvertToDuration(utils.GetEnv("TIMEOUT_READ", "")),
+				ReadHeader: utils.ConvertToDuration(utils.GetEnv("TIMEOUT_READ_HEADER", "")),
+				Write:      utils.ConvertToDuration(utils.GetEnv("TIMEOUT_WRITE", "")),
+				Idle:       utils.ConvertToDuration(utils.GetEnv("TIMEOUT_IDLE", "")),
+				Handler:    utils.ConvertToDuration(utils.GetEnv("TIMEOUT_HANDLER", "")),
+			},
+			Forwarding: Forward{
+				Host:               utils.GetEnv("FORWARD_HOST", ""),
+				Port:               utils.GetEnv("FORWARD_PORT", ""),
+				Scheme:             utils.GetEnv("FORWARD_SCHEME", ""),
+				Endpoints:          strings.Split(utils.GetEnv("LB_ENDPOINT_LIST", ""), ","),
+				HTTP2HTTPS:         utils.GetEnv("HTTP2HTTPS", "") == "1",
+				RedirectStatusCode: utils.ConvertToInt(utils.GetEnv("REDIRECT_STATUS_CODE", "")),
+			},
+			GZip: utils.GetEnv("GZIP_ENABLED", "") == "1",
+		},
+		Cache: Cache{
+			Host:            utils.GetEnv("REDIS_HOST", ""),
+			Port:            utils.GetEnv("REDIS_PORT", ""),
+			Password:        utils.GetEnv("REDIS_PASSWORD", ""),
+			DB:              utils.ConvertToInt(utils.GetEnv("REDIS_DB", "")),
+			TTL:             utils.ConvertToInt(utils.GetEnv("DEFAULT_TTL", "")),
+			AllowedStatuses: utils.ConvertToIntSlice(strings.Split(utils.GetEnv("CACHE_ALLOWED_STATUSES", ""), ",")),
+			AllowedMethods:  strings.Split(utils.GetEnv("CACHE_ALLOWED_METHODS", ""), ","),
+		},
+	}
 }
 
-// InitConfigFromFileOrEnv - Init the configuration in sequence: from a YAML file, from environment variables,
-// then defaults.
-func InitConfigFromFileOrEnv(file string) {
-	Config = Configuration{}
+func getYamlConfig(file string) Configuration {
+	YamlConfig := Configuration{}
```

Comment:
since it is using Configuration, document how to configure CircuitBreaker and TLS.Override

Reply:
done